### DISCUSSION TITLE
Get all entities considering xclip-clipping area

### DISCUSSION THREAD

**Discussion by cococosusu** [mozman/ezdxf#1084] [Q&A] [answered]
Hi!

I have been using the features of the recently released version 1.3.0 and they are very helpful.
Thank you.

I have a question regarding the existing **recursive_decompose** function.
When using this function, all entities are decomposed recursively, but as mentioned in the documentation, the clipping paths of xclip are not considered.

So, my question is:
How should I approach applying clipping paths to the decomposed entities?

I would appreciate any ideas you might have.

Thank you in advance ;)

**Comment by mozman** [comment]
Sorry, I can't help - it just don't work.

**Comment by mozman** [comment, marked as answer]
There is a way to explode clipped block references.

It is not possible to clip arbitrary DXF entities with `ezdxf` but it is possible by using the `drawing` add-on and the `DXFBackend`.
This process converts all entities inside the clipped block reference into drawing primitives. The clipping path is applied to the drawing primitives and the result is exported as DXF entities. 

These new DXF entities are not related to the input entities! The layer is preserved but everything else is lost e.g. dashed lines will be exploded into short solid lines, similar to importing PDF files.

```python
import ezdxf
from ezdxf import xclip
from ezdxf.addons.drawing import Frontend, RenderContext, dxf


def main():
    doc = ezdxf.readfile("xclip.dxf")
    msp = doc.modelspace()
    backend = dxf.DXFBackend(msp)  # target layout msp could be a new BLOCK
    frontend = Frontend(RenderContext(doc), backend)
    for insert in msp.query("INSERT"):
        if xclip.XClip(insert).is_clipping_enabled:  
            frontend.draw_entities([insert])
            insert.destroy()
    doc.saveas("exploded.dxf")


if __name__ == "__main__":
    main()
```
input file:  [xclip.zip](https://github.com/mozman/ezdxf/files/15428920/xclip.zip)

As an example the `CIRCLE` is converted to a `POLYLINE`:

![image](https://github.com/mozman/ezdxf/assets/4230966/5db25c7e-a30d-497e-8baf-4add9b2c0dce)

**Reply by cococosusu** [reply]
OMG...it's so perfect!
This is exactly the result I was hoping for.

I understood what was said well, and I will refer to it and use it well!

Have a happy Friday :)
Thank you so much :+1:
